src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountSyncUtils.java

```diff
@@ -0,0 +1,71 @@
+package com.commercetools.sync.sdk2.cartdiscounts.utils;
+
+import static com.commercetools.sync.sdk2.cartdiscounts.utils.CartDiscountUpdateActionUtils.*;
+import static com.commercetools.sync.sdk2.commons.utils.CustomUpdateActionUtils.buildPrimaryResourceCustomUpdateActions;
+import static com.commercetools.sync.sdk2.commons.utils.OptionalUtils.filterEmptyOptionals;
+
+import com.commercetools.api.models.cart_discount.CartDiscount;
+import com.commercetools.api.models.cart_discount.CartDiscountDraft;
+import com.commercetools.api.models.cart_discount.CartDiscountUpdateAction;
+import com.commercetools.sync.sdk2.cartdiscounts.CartDiscountSyncOptions;
+import com.commercetools.sync.sdk2.cartdiscounts.helpers.CartDiscountCustomActionBuilder;
+import com.commercetools.sync.sdk2.cartdiscounts.models.CartDiscountCustomTypeAdapter;
+import com.commercetools.sync.sdk2.cartdiscounts.models.CartDiscountDraftCustomTypeAdapter;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+public final class CartDiscountSyncUtils {
+
+  private static final CartDiscountCustomActionBuilder cartDiscountCustomActionBuilder =
+      new CartDiscountCustomActionBuilder();
+
+  /**
+   * Compares all the fields of a {@link io.sphere.sdk.cartdiscounts.CartDiscount} and a {@link
+   * io.sphere.sdk.cartdiscounts.CartDiscountDraft}. It returns a {@link java.util.List} of {@link
+   * io.sphere.sdk.commands.UpdateAction}&lt;{@link io.sphere.sdk.cartdiscounts.CartDiscount}&gt; as
+   * a result. If no update action is needed, for example in case where both the {@link
+   * io.sphere.sdk.cartdiscounts.CartDiscount} and the {@link
+   * io.sphere.sdk.cartdiscounts.CartDiscountDraft} have the same fields, an empty {@link
+   * java.util.List} is returned.
```

Comment:
```suggestion
   * Compares all the fields of a {@link CartDiscount} and a {@link
   * CartDiscountDraft}. It returns a {@link java.util.List} of {@link
   * CartDiscountUpdateAction} as
   * a result. If no update action is needed, for example in case where both the {@link
   * CartDiscount} and the {@link
   * CartDiscountDraft} have the same fields, an empty {@link
   * java.util.List} is returned.
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1016/commits/78909c7288f5d5e9cb250af14a4f1e16c6e0a5be